pkg/platform/packet/packet.go

```diff
@@ -451,6 +452,32 @@ func (c *config) checkReservationIDs() hcl.Diagnostics {
 	return diagnostics
 }
 
+// checkIPXEScriptURL checks that if os_version is specified then
+// ipxe_script_url should be present as os_version only works with IPXE.
+func (c *config) checkIPXEScriptURL() hcl.Diagnostics {
+	var diagnostics hcl.Diagnostics
+
+	if c.OSVersion != "" && c.IPXEScriptURL == "" {
+		diagnostics = append(diagnostics, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "cannot find the ipxe_script_url",
```

Comment:
```suggestion
			Summary:  "ipxe_script_url not specified",
```

"cannot find" sounds strange to me. "not specified" makes it clearer that the user needed to specify something but didn't.